### PR TITLE
Refactor autopilot package

### PR DESCRIPTION
**Pull request by patapufate**
This patch:
- Remove unused variables - Obsolet/Unused
- Remove unused file - Obsolete/Unused code.
- Remove unused messages and services files - Obsolete/Unused code.
- Split autopilot.cpp into header and sources files.
- Remove unused parameters and convert minimal vehicle speed constant to parameter.
- Remove unnecessary boilerplate in CMakeList.txt and Package.xml.
- Add mantainer and license.
- Add Roslint and fix linter errors
- Change folder structure for header files. Now implemented as "include/<package_name>/header_file.h"

It needs thorough testing, as I don't have the hardware at hand.

### CODE REVIEW

File: autopilot/src/autopilot_main.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  AutoPilotNode autoPilotNode(nh);
+  autoPilotNode.Start();
+
+  ros::spin();
+
+  autoPilotNode.Stop();
+
+  return (0);
+}
```
Comment (hidmic) [inline autopilot/src/autopilot_main.cpp:58]: @patapufate nit: missing newline.

File: autopilot/include/autopilot.h [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+  double desiredRoll;
+  double desiredPitch;
+  double desiredYaw;
+  double desiredDepth;
+  double desiredRudder;
+  double desiredSpeed;
+
+  double maxAllowedThrusterRpm;
+  double minimalspeed;  // knots
```
Comment (hidmic) [inline autopilot/include/autopilot.h:161]: @patapufate I see a mixture of camel case and snake case variable naming. Should we normalize?

**Review by hidmic — APPROVED** [review]
Overall LGTM !

File: autopilot/include/autopilot.h (hunk 2 above)
Comment (patapufate) [reply autopilot/include/autopilot.h:161]: @ChrisScianna do you have any preference on this?

**Review by hidmic — APPROVED** [review]
LGTM

**Pull request merged**